Model/Tax/Sales/Total/Quote/Tax.php

```diff
@@ -119,9 +128,17 @@ public function collect(
             $quote->getStoreId()
         );
 
-        if ($isEnabled) {
+        /**
+         * This registry check is to resolve bug https://github.com/magento/magento2/issues/9580
+         * In case of infinite loop we make sure the taxjar api is only hit once to avoid billing customers 
```

Comment:
Let's drop "to avoid billing customers for accidental transactions." to keep this comment to 2 lines. Thx!